### ISSUE TITLE
Contacts list shows up empty

### ISSUE THREAD

**Issue by TAHassing** [Dax89/harbour-sailorgram#26]
Hey :) Telegram app works great. However, if I open up the Contacts list through the pull-down menu, it shows up completely empty (despite the fact that I already have 5 conversations going).

I've been using SailorGram since v 71-1, and the problem persists in v72-1. Before upgrading to 72-1, I removed SailorGram 71-1 through the Warehouse, manually removed the ~/.local/share/harbour-sailorgram folder and rebooted the phone.

**Comment by Dax89** [comment]
Hi!
There is a new "Debug" option in SailorGram 0.72
Can you tell me how many contacts sees the client there? :)

**Comment by TAHassing** [comment]
Yeah that's me indeed. If I look under Debug, it says: heartbeat connected, 0 contacts.

Is there anything I can do to further investigate what's going on? Perhaps some analysis of my contacts DB or something?

**Comment by Dax89** [comment]
That info comes directly from Cutegram's libraries.

I have saw this behavior when someone contacted me using Telegram by I don't have it in my contact list (Telegrams treats it as "Foreign contacts")
There should be available an option "Add to contacts", you can find it by opening a conversation and swyping in the attached page on the right.

About DB, yes there is one, but it's filled by the information that comes from Telegram servers, in this case you are not receiving contact's information, so it is empty

**Comment by TAHassing** [comment]
This morning I checked the Error log in the Debug section and noticed an error regarding to 'AuthenticationKey' and 'contactsGetContacts' or similar. I don't know exactly what it said anymore, because it seems the app cleared the error log, so it's gone now.

I've tried to reproduce the error by removing the /.local/share/harbour-sailorgram folder and re-do the login authentication, but the error log is still empty. I'll continue to try to reproduce the bug

**Comment by Dax89** [comment]
Ok!
I have began the development of a dedicated Telegram library, so I can get rid off the one used in 0.72 because contains bugs and it's heavy for a mobile phone.

Probably this new library fixes you issue too :)

**Comment by TAHassing** [comment]
You're building your own library? That's really cool! Good luck with the development, really curious to see how it will turn out :)

**Comment by Dax89** [comment]
Yes, it's easier to maintain the software and make bugfixing (for me, at least) :)

**Comment by lewadedun** [comment]
for me Contacts list empty  too )

**Comment by Dax89** [comment]
Is this bug still present in 0.73? :)

**Comment by TAHassing** [comment]
Since my previous comment I've installed the Android version of Telegram (official client) once and it seems to have filled my contact list. After that, I removed the Android version and reinstalled Sailorgram. The contact list shows up correctly in Sailorgram now too, also with freshly added contacts ;) So I don't know whether this is due to a bug fix in Sailorgram or otherwise.

**Comment by Dax89** [comment]
Mmmh maybe there is some way to sync it, I'll keep this issue open as reminder :)
